rewrite scale controller

denis-emanuel opened the pull request:
add static weight variable to scale controller
rewrite old read method of scale controller to update and accept number of units as parameter

denis-emanuel committed:
add static weight variable to scale controller
rewrite old read method of scale controller to update and accept number of units as parameter

Bartie5 reviewed (changes requested):
Schimbarile la obiect sunt bune, dar trebuie apelata in main loop metoda de update.

Propun sa folosim pt inceput un weightUpdateFlag (similar cu lcdUpdateFlag). lcdUpdateFlag trebuie facut true in continuu in modul de POURING undeva. 
In modul AUTO, as propune sa facem flag-ul true o data la 1000ms, folosind un weightReadingCounter si un WEIGHT_READ_MS. 

Valoarea lui READ_SCALE_MS nu mai e nevoie sa fie 120ms. O putem micsora la 10/20ms.

Pentru apelul de update weight, as incerca cu update(1). In functie de cum se comporta, mai vedem dupa cum punem.

denis-emanuel commented on include/scale.h:
```diff
@@ -8,10 +8,15 @@ class ScaleController
 private:
     HX711 scale;
-    unsigned int readingCounter = 0;
+    static unsigned int readingCounter;
```
dasbdasb

denis-emanuel commented on include/scale.h:
```diff
@@ -8,10 +8,15 @@ class ScaleController
 public:
-    ScaleController() {}
+    ScaleController()
+    {
```
dsadas

denis-emanuel committed:
finish scale refactoring and setup

denis-emanuel committed:
add recipes for aperol spritz and mojito
change logic to use empty pumps first if a new recipe is added and the ingredient is not found anywhere else
cleanup visual for switch case statements
map potentiometer value to number of ingredients not to number of pumps for preview mode
for preview mode show each ingredient of the recipe and on which pump it belongs
